src/utils/fetch-utils.js

```diff
@@ -1,25 +1,39 @@
 export const doFetch = (url, method='GET', body, headers={
-    'Content-Type': 'application/json'    
-}) => {   
+    'Content-Type': 'application/json'
+}) => {
     const plomise = new Promise((resolve, reject)=>{
         fetch(url, {
             method,
             body: JSON.stringify(body),
             headers
         })
-            .then((res) => {            
-                res.json()
-                    .then((json)=>{
-                        if(res.ok){ //200 level error
-                            resolve(json);
-                        } else {
-                            reject(json); //fail with error response from server
-                        }
-                    })
-                    .catch((e)=>reject(e)); //error processing json response      
+            .then((res) => {
+                if (res.status === 204){
+                    resolve();
+                } else {
+                    res.json()
+                        .then((json)=>{
+                            if(res.ok){ //200 level error
+                                resolve(json);
+                            } else {
+                                reject(json); //fail with error response from server
+                            }
+                        })
+                        .catch((e)=>reject(e)); //error processing json response
+                }
             })
-            .catch((e)=>reject(e)); //fetch error   
+            .catch((e)=>reject(e)); //fetch error
     })
     
     return plomise;
-};
\ No newline at end of file
+};
+
+
+export const doGet = (url, body) => doFetch(url, 'POST', body)
```

Comment:
whats the utility of this?

Reply:
mimic a 404 response one would expect from a GET if the item does not exist

Reply:
also resolve with the first item in the search results array, instead of having to do that over and over